app/assets/javascripts/discourse/app/components/time-shortcut-picker.js

```diff
@@ -0,0 +1,250 @@
+  },
+  "n d": {
+    handler: "selectShortcut",
+    args: [TIME_SHORTCUT_TYPES.TOMORROW],
+  },
+  "n w": {
+    handler: "selectShortcut",
+    args: [TIME_SHORTCUT_TYPES.NEXT_WEEK],
+  },
+  "n b w": {
+    handler: "selectShortcut",
+    args: [TIME_SHORTCUT_TYPES.START_OF_NEXT_BUSINESS_WEEK],
+  },
+  "n m": {
+    handler: "selectShortcut",
+    args: [TIME_SHORTCUT_TYPES.NEXT_MONTH],
+  },
+  "c r": { handler: "selectShortcut", args: [TIME_SHORTCUT_TYPES.CUSTOM] },
+  "n r": { handler: "selectShortcut", args: [TIME_SHORTCUT_TYPES.NONE] },
+};
+
+export default Component.extend({
+  tagName: "",
+
+  userTimezone: null,
+
+  onTimeSelected: null,
+
+  selectedShortcut: null,
+  selectedTime: null,
+  selectedDate: null,
+  selectedDatetime: null,
+  prefilledDatetime: null,
+
+  additionalOptionsToShow: [],
+  hiddenOptions: [],
+  customOptions: [],
+
+  lastCustomDate: null,
+  lastCustomTime: null,
+  parsedLastCustomDatetime: null,
+  customDate: null,
+  customTime: null,
+
+  defaultCustomReminderTime: `0${START_OF_DAY_HOUR}:00`,
+
+  @on("init")
+  _setupPicker() {
+    this.setProperties({
+      customTime: this.defaultCustomReminderTime,
+      userTimezone: this.currentUser.resolvedTimezone(this.currentUser),
+    });
+
+    if (this.prefilledDatetime) {
+      let parsedDatetime = parseCustomDatetime(
+        this.prefilledDatetime,
+        null,
+        this.userTimezone
+      );
+
+      if (parsedDatetime.isSame(laterToday())) {
+        return this.set("selectedShortcut", TIME_SHORTCUT_TYPES.LATER_TODAY);
+      }
+
+      this.setProperties({
+        customDate: parsedDatetime.format("YYYY-MM-DD"),
+        customTime: parsedDatetime.format("HH:mm"),
+        selectedShortcut: TIME_SHORTCUT_TYPES.CUSTOM,
+      });
+    }
+
+    this._bindKeyboardShortcuts();
+    this._loadLastUsedCustomDatetime();
+  },
+
+  @on("willDestroyElement")
+  _resetKeyboardShortcuts() {
+    KeyboardShortcuts.unbind(BINDINGS);
+    KeyboardShortcuts.unpause(GLOBAL_SHORTCUTS_TO_PAUSE);
+  },
+
+  _loadLastUsedCustomDatetime() {
+    let lastTime = localStorage.lastCustomTime;
+    let lastDate = localStorage.lastCustomDate;
+
+    if (lastTime && lastDate) {
+      let parsed = parseCustomDatetime(lastDate, lastTime, this.userTimezone);
+
+      if (parsed < now(this.userTimezone)) {
+        return;
+      }
+
+      this.setProperties({
+        lastCustomDate: lastDate,
+        lastCustomTime: lastTime,
+        parsedLastCustomDatetime: parsed,
+      });
+    }
+  },
+
+  _bindKeyboardShortcuts() {
+    KeyboardShortcuts.pause(GLOBAL_SHORTCUTS_TO_PAUSE);
+    Object.keys(BINDINGS).forEach((shortcut) => {
+      KeyboardShortcuts.addShortcut(shortcut, () => {
+        let binding = BINDINGS[shortcut];
+        if (binding.args) {
+          return this.send(binding.handler, ...binding.args);
+        }
+        this.send(binding.handler);
+      });
+    });
+  },
+
+  @discourseComputed("selectedShortcut")
```

Comment:
could probably use propertyEqual

Reply:
How though, `TIME_SHORTCUT_TYPES.CUSTOM` is not a property?

Reply:
Sorry I meant equal: https://api.emberjs.com/ember/2.16/functions/@ember%2Fobject%2Fcomputed/equal

Reply:
Ah OK, will change